### ISSUE TITLE
Query refuses to execute

### ISSUE THREAD

**Issue by lyziane** [cakephp/cakephp#11091]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.4.2
* Platform and Target: Running local server with Wamp on Windows 10

### What you did
In my `ChatChannel` entity I wrote a function `getOrCreateChatUser($data)` were I pass in some data with the goal of getting my `ChatUser` entity related to that data. The data can be a `User` entity, a `Customer` entity, a `ChatUser` array (to retrieve entity from id) or else it will use the Authenticated User data. So I started my query at the top of my function and depending on all the If-Else conditions I pick it back up and add my conditions and run it.

### What happened
Even if I call `->first()` on the query it refuses to execute.

### Here's some code
Basically if I use my function to find my `ChatUser` with my `Customer` entity...

If my query is written in one go, it works:

```
$chatUser = $ChatChannels->ChatMembers->ChatUsers->find()
->where(['customer_id' => $data->id])
->first();
```

BUT if I cut it in multiple lines then it cannot be executed:

```
$chatUser = $ChatChannels->ChatMembers->ChatUsers->find();
			
if (is_a($data, 'App\Model\Entity\User'))
	$chatUser->where(['user_id' => $data->id]);
if (is_a($data, 'App\Model\Entity\Customer'))
	$chatUser->where(['customer_id' => $data->id]);
			
$chatUser->first();
```

### Why?
I don't know why but I wrote queries in multiple lines like this before and everything was good. Is it a because it's a query launched from inside an entity function?

**Comment by AD7six** [comment]
what does "it cannot be executed" mean?

Irrespective, I don't see how this is likely to be a framework problem

**Comment by lyziane** [comment]
I'm using the keywords from the CakePHP documentation. Executing a query is when the query object is run as a SQL statement.

**Example:**

This creates a Query object:
`$chatUser = $ChatChannels->ChatMembers->ChatUsers->find();`

This executes it:
`$chatUser->first()`

You can also use `->all()` or `->toArray()` or `->execute()` (not sure what this one does differently, I never use it)

You can choose to ignore my issue if you want and I'll just copy-paste the same query 4 times but I assume that when you call a function from a framework and the expected result doesn't happen, it's a framework problem...

**Comment by lyziane** [comment]
So if I debug my same `$chatUser` from my examples:
```
object(Cake\ORM\Query) {

	'(help)' => 'This is a Query object, to get the results execute or iterate it.',
	'sql' => 'SELECT ChatUsers.id AS `ChatUsers__id`, ChatUsers.user_id AS `ChatUsers__user_id`, ChatUsers.customer_id AS `ChatUsers__customer_id`, ChatUsers.sid AS `ChatUsers__sid`, ChatUsers.identity AS `ChatUsers__identity`, ChatUsers.email AS `ChatUsers__email`, ChatUsers.created AS `ChatUsers__created`, ChatUsers.modified AS `ChatUsers__modified`, ChatUsers.closed AS `ChatUsers__closed` FROM chat_users ChatUsers WHERE customer_id = :c0 LIMIT 1',
	...
}
```
The SQL query is prepared properly but there's no way to make it run and actually get my entity (or false)

**Comment by AD7six** [comment]
Well, this:
```
$chatUser = $ChatChannels->ChatMembers->ChatUsers->find()
->where(['customer_id' => $data->id])
->first();
```

Is not equivalent to this:

```
$chatUser = $ChatChannels->ChatMembers->ChatUsers->find();			
...
$chatUser->first();
```

In the first case `$chatUser` is the result of calling `first` on an intermediary query object. I.e. the result is an entity or falsey.

In the second case `$chatUser` is a query object, and the result of calling `first` is not used. 

Calling `first` does not modify the query object, it [returns the first result](https://github.com/cakephp/cakephp/blob/master/src/Collection/CollectionTrait.php#L365-L374)

This appears to be your intent:
```
$query = $ChatChannels->ChatMembers->ChatUsers->find();
...
$chatUser = $query->first()
```

Closing as this is a misunderstanding.

**Comment by lyziane** [comment]
Thank you @AD7six. I feel really dumb now... haha
I guess I just assumed that you can only execute your query object once, I should have looked at what `->first()` does...
